src/azure_pdf_parser/utils.py

```diff
@@ -71,6 +71,11 @@ def propagate_page_number(batch: PDFPagesBatchExtracted) -> PDFPagesBatchExtract
                     bounding_region.page_number = (
                         bounding_region.page_number + page_offset
                     )
+
+    if batch.extracted_content.pages:
```

Comment:
nit: this `if` isn't required, but we can fix that in the follow-up to add a regression test